lib/orogen/gen/tasks.rb

```diff
@@ -533,6 +533,20 @@ def generate
                     hidden_operation("__orogen_getTID", "    #ifdef HAS_GETTID\nreturn syscall(SYS_gettid);\n#else\nreturn 0;\n#endif").
                         returns("int").
                         doc("returns the PID for this task")
+
+                    if extended_state_support?
```

Comment:
The name of the states is already stored in the typelib registry (as the project_name/TaskName_TASK_STATES enum). Given the cost (in terms of compilation time) of adding a new operation, I would really avoid that.